src/Client/PullPayment.php

```diff
@@ -0,0 +1,274 @@
+        bool $includeArchived
+    ): \BTCPayServer\Result\PullPaymentList {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $body = json_encode(
+            [
+                'includeArchived' => $includeArchived,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\PullPaymentList(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function createPullPayment(
+        string $storeId,
+        ?string $name = null,
+        PreciseNumber $amount,
+        string $currency,
+        ?int $period,
+        ?int $BOLT11Expiration,
+        ?int $startsAt,
+        ?int $expiresAt,
+        array $paymentMethods
+    ): \BTCPayServer\Result\PullPayment {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'name' => $name,
+                'amount' => $amount->__toString(),
+                'currency' => $currency,
+                'period' => $period,
+                'BOLT11Expiration' => $BOLT11Expiration,
+                'startsAt' => $startsAt,
+                'expiresAt' => $expiresAt,
+                'paymentMethods' => $paymentMethods
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\PullPayment(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function archivePullPayment(
+        string $storeId,
+        string $pullPaymentId
+    ): bool {
+        $url = $this->getApiUrl() . 'stores/' .
+        urlencode($storeId) . '/' . 'pull-payments/' .
+        urlencode($pullPaymentId);
+
+        $headers = $this->getRequestHeaders();
+        $method = 'DELETE';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function approvePayout(
+        string $storeId,
+        string $payoutId,
+        int $revision,
+        mixed $rateRule
```

Comment:
@barrydeen the `mixed` type does not exist, we need to have something more specific here, my guess is "string"? If you want to make it an optional argument add `?` so `?string`

Reply:
Fixed